pkg/transform/jq.go

```diff
@@ -169,3 +171,16 @@ func mkJQInput(jqm *jqMapper, message *models.Message, interState interface{}) (
 
 	return spInput, nil
 }
+
+func removeNullFields(data any) {
+	if asMap, ok := data.(map[string]interface{}); ok {
+		for key := range asMap {
```

Comment:
If I read it right, this iterates the top level fields of the map, and if they're null it removes them. But we already have both of these structures:

```
{
    "attributes": {
        "someKey": null
    }
}
```

```
{
    "attributes": [
        {
           "someKey": null
        }
    ],
}
```

So I think it would need to recursively iterate and deal with both elements of slices and keys of maps.

I think this might become quite complex - I wonder if there's something clever that we can do in the templater to solve this more elegantly?

Reply:
You're right, I forgot about arrays... 🤦

Reply:
Pushed another commit, looks better now?